test/rustic-compile-test.el

```diff
@@ -85,12 +85,25 @@
 
 (ert-deftest rustic-test-compile ()
   (let* ((dir (rustic-babel-generate-project t)))
-    (should-not compilation-directory)
-    (should-not compilation-arguments)
-    (setq compilation-arguments "cargo fmt")
+    (let* ((default-directory dir)
+           (proc (rustic-compile "cargo fmt")))
+      (should (process-live-p proc))
+      (while (eq (process-status proc) 'run)
+        (sit-for 0.1))
+      (should (string= compilation-directory dir))
+      (let ((proc (rustic-recompile)))
```

Comment:
Why are we doing `rustic-recompile` here ? From a quick look the previous code doesn't seem to have it.

Reply:
The previous code does have it

https://github.com/emacs-rustic/rustic/blob/0d79e4c19b7b77960a7665b8cef181954e4023e5/test/rustic-compile-test.el#L98

Reply:
I added a new test for compiling interactively rather than via lisp, so the diff looks a little weird.